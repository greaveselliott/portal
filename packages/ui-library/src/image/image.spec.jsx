import React from 'react';
import Image from './image';
import { shallow } from 'enzyme';

describe('Image component...', () => {
    const defaultProps = { className: '', height: null, src: '', width: null };
    const render = (props = defaultProps) => shallow(<Image {...props} />);

    it('should render correctly', () => {
        const SUT = render({
            className: 'abc',
            height: 123,
            src: 'abc',
            width: 123
        });
        
        expect(SUT).toMatchSnapshot();
    });
});