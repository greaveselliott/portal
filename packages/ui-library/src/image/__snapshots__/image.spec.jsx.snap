// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className="abc"
    height={123}
    src="abc"
    width={123}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "eg-image abc",
      "height": 123,
      "src": "abc",
      "width": 123,
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "eg-image abc",
        "height": 123,
        "src": "abc",
        "width": 123,
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
